src/ui/public/vis_maps/maps_renderbot.js

```diff
@@ -230,13 +230,45 @@ module.exports = function MapsRenderbotFactory(Private, $injector, serviceSettin
       };
     }
 
-    _doRenderComplete() {
+    _doRenderCompleteWhenBaseLayerIsLoaded(resolve, endTime) {
       if (this._paramsDirty || this._dataDirty || this._baseLayerDirty) {
-        return;
+        if (Date.now() <= endTime) {
+          setTimeout(() => {
```

Comment:
It seems like we could be using (endTime - Date.now()) to only do one setTimeout that fires the renderComplete if we haven't yet. I'm not understanding why we continually call `_doRenderCompleteWhenBaseLayerIsLoaded` every 10 ms recursively...